### DISCUSSION TITLE
Can't reach database server vercel & supabase

### DISCUSSION THREAD

**Discussion by Mujhtech** [prisma/prisma#21741] [Q&A] [answered]
### Question

Please does anyone knows what likely to be an issue here is what happened after deploy to Vercel this error used to come up every time like once you load the webpage the error will show then once you reload it will work normal if you try reload again may be in the next 1 minutes the error will show again rinse and repeat.

<img width="694" alt="Screenshot 2023-11-02 at 08 28 08" src="https://github.com/prisma/prisma/assets/41507005/a02f414b-9278-4abb-96e8-069508c05d2e">

Here is the log on Vercel

```json
{
"errorType":"Runtime.UnhandledPromiseRejection",
"errorMessage":"PrismaClientInitializationError: Can't reach database server at `aws-0-eu-west-2.pooler.supabase.com`:`6543`\n\nPlease make sure your database server is running at `aws-0-eu-west-2.pooler.supabase.com`:`6543`.",
"reason":{"errorType":"PrismaClientInitializationError","errorMessage":"Can't reach database server at `aws-0-eu-west-2.pooler.supabase.com`:`6543`\n\nPlease make sure your database server is running at `aws-0-eu-west-2.pooler.supabase.com`:`6543`.",
"name":"PrismaClientInitializationError","clientVersion":"5.4.1","errorCode":"P1001","stack":["PrismaClientInitializationError: Can't reach database server at `aws-0-eu-west-2.pooler.supabase.com`:`6543`","","Please make sure your database server is running at `aws-0-eu-west-2.pooler.supabase.com`:`6543`.","    at r (/var/task/node_modules/.pnpm/@prisma+client@5.4.1_prisma@5.4.1/node_modules/@prisma/client/runtime/library.js:114:2740)"]},"promise":{},"stack":["Runtime.UnhandledPromiseRejection: PrismaClientInitializationError: Can't reach database server at `aws-0-eu-west-2.pooler.supabase.com`:`6543`","","Please make sure your database server is running at `aws-0-eu-west-2.pooler.supabase.com`:`6543`.","    at process.<anonymous> (file:///var/runtime/index.mjs:1276:17)","    at process.emit (node:events:529:35)","    at emit (node:internal/process/promises:149:20)","    at processPromiseRejections (node:internal/process/promises:283:27)","    at process.processTicksAndRejections (node:internal/process/task_queues:96:32)"]}
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by janpio** [comment]
How long does the function run before it crashes with this error message?

Where are you running these serverless functions? The database connection string includes `eu-west-2`, so optimally the serverless functions would run close to that so that opening a connection to the database does not have to go around the globe.

**Reply by Mujhtech** [reply]
Oh, I actually deployed the web app on vercel no configuration just using default vercel configuration and you know vercel use serverless function

**Reply by janpio** [reply]
You should be able to configure the location of our serverless functions in the Vercel UI.
Or you can change your region in Supabase as well to match the default location of Vercel.

**Reply by Mujhtech** [reply]
This method seems to work, but I arrived at another issue. Basically, I can't change location after creating a database in Supabase, but I was able to change location in Vercel dashboard to London, UK. A new error I'm getting now is about ConnectionError. I tried to replicate it in localhost by replacing the database url with Supabase's own and I got same thing check out the video below

https://drive.google.com/file/d/1d2uLiC1NXLs7lw8IHEzj_h55LFr8jPZH/view?usp=sharing

**Reply by janpio** [reply, marked as answer]
Are you using the pooled connection from Supabase?
Then have a look at this: https://www.prisma.io/docs/guides/performance-and-optimization/connection-management/configure-pg-bouncer

**Reply by Mujhtech** [reply]
Yeah, thanks for this reminder. I just tested with pgbouncer, and it works. So what is that I set the pgbouncer before I started testing but removed it when I wanted to test because Supabase mentioned that pgbouncer is officially deprecated

**Reply by Mujhtech** [reply]
https://github.com/orgs/supabase/discussions/17817#top

**Reply by janpio** [reply]
Supavisor also required `pgbouncer=true` for now. We'll try to come up with a better name for that configuration.

**Comment by Made4Uo** [comment]
Have you tried adding "postinstall": "npm run generate" to package.json? That fix my issue
